source/connect/connection-options/server-selection.txt

```diff
@@ -1,68 +1,193 @@
-           Type: "Unknown",
-           HeartbeatException: "<exception details>"
-       }] 
-   }.
-
-The error message consists of multiple parts:
-
-1. The server selection timeout (30000 ms).
-#. The server selectors considered (``CompositeServerSelector``
-   containing ``AreSessionsSupportedServerSelector``,
-   ``LatencyLimitingServerSelector``, and
-   ``OperationsCountServerSelector``).
-#. The driver’s current view of the cluster topology. The list of
-   servers that the driver is aware of is a key part of this view. Each
-   server description contains an exhaustive description of its current
-   state including information about an endpoint, a server version, a
-   server type, and its current health state. If the server encounters issues in
-   reporting its health, ``HeartbeatException`` contains the exception from the
-   last failed heartbeat. Analyzing the ``HeartbeatException`` on each
-   cluster node can assist in diagnosing most server selection issues.
-   The following heartbeat exceptions are common:
-   
-   * ``No connection could be made because the target machine actively
-     refused it``: The driver cannot see this cluster node. This can be
-     because the cluster node has crashed, a firewall is preventing
-     network traffic from reaching the cluster node or port, or some other
-     network error is preventing traffic from being successfully routed to
-     the cluster node.
-   * ``Attempted to read past the end of the stream``: This error
-     happens when the driver cannot connect to the cluster nodes due to a
-     network error, misconfigured firewall, or other network issue. To
-     address this exception, ensure that all cluster nodes are reachable.
-     This error commonly occurs when the client machine’s IP address is
-     not configured in the Atlas IPs Access List, which can be found under
-     the :guilabel:`Network Access` tab for your Atlas Project.
-   * ``The remote certificate is invalid according to the validation
-     procedure``: This error typically indicates a TLS/SSL-related problem
-     such as an expired/invalid certificate or an untrusted root CA. You
-     can use tools like ``openssl s_client`` to debug TLS/SSL-related
-     certificate problems.
\ No newline at end of file
+==========================
+Customize Server Selection
+==========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example, read preference, write
+
+Overview
+--------
+
+All MongoDB drivers follow a defined algorithm when selecting a server to read or write
+from. By using the ``ClusterConfigurator`` property of a ``MongoClient`` object, you can
+customize this algorithm to choose the server that works best for your application.
+
+.. important::
+
+   Customizing the server-selection algorithm can have unintended consequences,
```

Comment:
[Use "might" to indicate possibility](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#std-term-might)
```suggestion
   Customizing the server-selection algorithm might have unintended consequences,
```